funsor/distributions.py

```diff
@@ -168,134 +172,89 @@ def _infer_param_domain(cls, name, raw_shape):
 # Distribution Wrappers
 ################################################################################
 
-def make_dist(pyro_dist_class, param_names=()):
 
-    if not param_names:
-        param_names = tuple(name for name in inspect.getfullargspec(pyro_dist_class.__init__)[0][1:]
-                            if name in pyro_dist_class.arg_constraints)
+def _eager_log_prob(dist_name, *params):
+    backend_dist_module = import_module(BACKEND_TO_DISTRIBUTIONS_BACKEND[get_backend()])
+    return getattr(backend_dist_module, dist_name).eager_log_prob(*params)
 
-    @makefun.with_signature(f"__init__(self, {', '.join(param_names)}, value='value')")
-    def dist_init(self, **kwargs):
-        return Distribution.__init__(self, *tuple(kwargs[k] for k in self._ast_fields))
 
-    dist_class = DistributionMeta(pyro_dist_class.__name__.split("_PyroWrapper_")[-1], (Distribution,), {
-        'dist_class': pyro_dist_class,
-        '__init__': dist_init,
-    })
+def make_dist(dist_name, param_names):
+    @makefun.with_signature(f"{dist_name}({', '.join(param_names)}, value='value')",
+                            eager_log_prob=functools.partial(_eager_log_prob, dist_name))
+    def dist_fn(*args, **kwargs):
+        backend_dist_module = import_module(BACKEND_TO_DISTRIBUTIONS_BACKEND[get_backend()])
+        return getattr(backend_dist_module, dist_name)(*args, **kwargs)
 
-    eager.register(dist_class, *((Tensor,) * (len(param_names) + 1)))(dist_class.eager_log_prob)
+    return dist_fn
 
-    return dist_class
 
+FUNSOR_DIST_NAMES = {
```

Comment:
Sure, we should do it if we use `import funsor.backend.distributions as dist` pattern. Let me think more if it is a good way. Right now I don't have to change the import statement, just simply `import funsor.distributions as dist`. But it does not matter much I guess.

Reply:
Maybe, for now, we just do what you suggested, and rename `funsor.distributions` to `funsor.distribution` to avoid confusion. We can decide what is a good way to do later.